database/seed.js

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable arrow-body-style */
+const mysql = require('mysql');
+const Promise = require('bluebird');
+const { seedAllData } = require('./seedMethods');
+
+
+const connection = mysql.createConnection({
+  user: 'root',
+  host: 'localhost',
+});
+
+const db = Promise.promisifyAll(connection, { multiArgs: true });
+
+const setupDb = () => {
+  return db.queryAsync(`
+    CREATE TABLE IF NOT EXISTS details (
+      id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+      type VARCHAR(20),
+      pagenum INT,
+      publisher VARCHAR(100),
+      firstPubDate VARCHAR(30),
+      originalPubDate VARCHAR(30),
+      title VARCHAR(100),
+      isbn10 VARCHAR(20),
+      isbn13 VARCHAR(20),
+      language VARCHAR(20)
+      );`)
+    .then(() => {
+      return db.queryAsync(`
+        CREATE TABLE IF NOT EXISTS characters (
+          id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+          name VARCHAR(100),
+          bookId INT
+        );`);
+    })
+    .then(() => {
+      return db.queryAsync(`
+        CREATE TABLE IF NOT EXISTS settings (
+          id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+          city VARCHAR(150),
+          country VARCHAR(150),
+          bookId INT
+        );`);
+    })
+    .then(() => {
+      return db.queryAsync(`
+        CREATE TABLE IF NOT EXISTS awards (
+          id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+          name VARCHAR(100),
+          year INT,
+          bookId INT
+        );`);
+    })
+    .then(() => {
+      return db.queryAsync(`
+        CREATE TABLE IF NOT EXISTS editions (
+          id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+          isbn10 VARCHAR(20),
+          isbn13 VARCHAR(20),
+          title VARCHAR(100),
+          type VARCHAR(20),
+          publisher VARCHAR(100),
+          originalPubDate VARCHAR(30),
+          coverurl VARCHAR(250),
+          bookId INT
+        );`);
+    })
+    .error((err) => {
```

Comment:
Can you tell me what is the motivation behind creating the db tables in the seed.js file and do you think cascading of promises is a good programming practice I would rather let things run independently?

Reply:
I created tables in the seed.js to exclude installing a mysql client within my circleci environment. This made it easier for me to setup my circleci environment. 

If I want to seed my database with by running the command mysql -u root < schema.sql within the npm seed command, I needed to install mysql-client within my circleci environment which made setting up the circleci environment harder. It was easier to simply use only a node command since node.js is my top dependancy in my circleci environment.

As for cascading promises, you're right that it would be more efficient to create the tables concurrently using a Promise.all method. However, I think I'll leave that in my backlogs to change later for now since there is not much difference other than taking a little longer during the seeding process when a promise chain is used.

Reply:
Ok, I see. Thanks for answering!